pkg/server/cloud/gce/instances.go

```diff
@@ -0,0 +1,445 @@
+	// Not supported in GCE return nil
+	return nil
+}
+
+func (c *gceClient) ListInstancesFilterID(ids []string) ([]cloud.CloudInstance, error) {
+	instances, err := c.ListInstances()
+	if err != nil {
+		return nil, err
+	}
+	var filteredList []cloud.CloudInstance
+	find := func(id string) {
+		for _, ci := range instances {
+			if ci.ID == id {
+				filteredList = append(filteredList, ci)
+			}
+		}
+	}
+	for _, id := range ids {
+		find(id)
+	}
+	return filteredList, nil
+}
+
+func (c *gceClient) ListInstances() ([]cloud.CloudInstance, error) {
+	listCall := c.service.Instances.List(c.projectID, c.zone)
+	filter := c.getLabelCompareFilter(controllerLabelKey, c.controllerID)
+	listCall = listCall.Filter(filter)
+	var instances []cloud.CloudInstance
+	f := func(page *compute.InstanceList) error {
+		for _, instance := range page.Items {
+			instances = append(instances, cloud.CloudInstance{
+				ID:       instance.Name,
+				NodeName: instance.Labels["node"],
+			})
+		}
+		return nil
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	if err := listCall.Pages(ctx, f); err != nil {
+		return nil, err
+	}
+	return instances, nil
+}
+
+func (c *gceClient) AddInstanceTags(iid string, labels map[string]string) error {
+	// in GCE "labels" are what AWS considers tags
+	labels = convertLabelKeys(labels)
+
+	// Todo, pull this from short term instance cache
+	inst, err := c.getInstanceSpec(iid)
+	if err != nil {
+		return util.WrapError(err, "error retrieving instance's label fingerprint from GKE")
+	}
+	mergedLabels := inst.Labels
+	for k, v := range labels {
+		mergedLabels[k] = v
+	}
+
+	labelRequest := compute.InstancesSetLabelsRequest{
+		LabelFingerprint: inst.LabelFingerprint,
+		Labels:           mergedLabels,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	op, err := c.service.Instances.SetLabels(c.projectID, c.zone, iid, &labelRequest).Context(ctx).Do()
+	if err != nil {
+		return util.WrapError(err, "Error attaching instance labels %s", err)
+	}
+	if err := c.waitOnOperation(op.Name, c.getZoneOperation); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *gceClient) GetImageID(spec cloud.BootImageSpec) (string, error) {
+	project, ok := spec["project"]
+	if !ok {
+		return "", fmt.Errorf("project is a required boot image value. Please specify cells.bootImageSpec.project in provider.yaml")
+	}
+	image, ok := spec["image"]
+	if !ok {
+		return "", fmt.Errorf("image is a required boot image value. Please specify cells.bootImageSpec.image in provider.yaml")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	resp, err := c.service.Images.Get(project, image).Context(ctx).Do()
+	if err != nil {
+		return "", util.WrapError(err, "Error looking up boot image %s/%s", project, image)
+	}
+	if resp == nil {
+		return "", nilResponseError("Images.Get")
+	}
+	return resp.SelfLink, nil
+}
```

Comment:
This will need a simple change since I've change the cloud API GetImageID() -> GetImage() to save cloud-specific image attributes that are needed when starting a new instance (we use it on AWS to save the disk device used by the image).

Reply:
made updates for this in 8fe2c2c